packages/rest/src/validation/request-query.validator.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  RequestBodyObject,
+  SchemaObject,
+  ReferenceObject,
+  SchemasObject,
+} from '@loopback/openapi-v3-types';
+import * as AJV from 'ajv';
+import * as debugModule from 'debug';
+import * as util from 'util';
+import {HttpErrors, RestHttpErrors, RequestBody} from '..';
+import * as _ from 'lodash';
+
+const toJsonSchema = require('openapi-schema-to-json-schema');
+const debug = debugModule('loopback:rest:validation');
+
+export type RequestQueryValidationOptions = AJV.Options;
+
+/**
+ * Check whether the request query is valid according to the provided OpenAPI schema.
+ * The JSON schema is generated from the OpenAPI schema which is typically defined
+ * by `@requestQuery()`.
+ * The validation leverages AJS schema validator.
+ * @param query The request query parsed from an HTTP request.
+ * @param requestQuerySpec The OpenAPI requestQuery specification defined in `@requestQuery()`.
+ * @param globalSchemas The referenced schemas generated from `OpenAPISpec.components.schemas`.
+ */
+export function validateRequestQuery(
+  query: RequestBody,
+  requestQuerySpec?: RequestBodyObject,
+  globalSchemas: SchemasObject = {},
+  options: RequestQueryValidationOptions = {},
+) {
+  const required = requestQuerySpec && requestQuerySpec.required;
+
+  if (required && query.value == undefined) {
+    const err = Object.assign(
+      new HttpErrors.BadRequest('Request query is required'),
+      {
+        code: 'MISSING_REQUIRED_PARAMETER',
+        parameterName: 'request query',
```

Comment:
It would be nice to see which `query` param.

Reply:
is this sufficient?
![query_validation_errors](https://user-images.githubusercontent.com/45032706/52175388-bb7e0580-27ab-11e9-8f44-5cd78671d998.JPG)

Reply:
Sure.